environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -0,0 +1,154 @@
+package com.sequenceiq.environment.environment.service;
+
+import static com.sequenceiq.environment.environment.EnvironmentStatus.AVAILABLE;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.BadRequestException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudRegions;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.util.ValidationResult;
+import com.sequenceiq.environment.CloudPlatform;
+import com.sequenceiq.environment.api.v1.environment.model.request.EnvironmentChangeCredentialRequest;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.environment.credential.domain.Credential;
+import com.sequenceiq.environment.credential.service.CredentialService;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.domain.Region;
+import com.sequenceiq.environment.environment.dto.EnvironmentDtoConverter;
+import com.sequenceiq.environment.environment.dto.EnvironmentEditDto;
+import com.sequenceiq.environment.environment.dto.LocationDto;
+import com.sequenceiq.environment.environment.repository.EnvironmentRepository;
+import com.sequenceiq.environment.network.NetworkService;
+import com.sequenceiq.environment.network.domain.BaseNetwork;
+import com.sequenceiq.environment.network.dto.NetworkDto;
+
+@Service
+public class EnvironmentModificationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnvironmentModificationService.class);
+
+    private final EnvironmentDtoConverter environmentDtoConverter;
+
+    private final EnvironmentRepository environmentRepository;
+
+    private final EnvironmentService environmentService;
+
+    private final ConversionService conversionService;
+
+    private final CredentialService credentialService;
+
+    private final NetworkService networkService;
+
+    public EnvironmentModificationService(EnvironmentDtoConverter environmentDtoConverter, EnvironmentRepository environmentRepository,
+            EnvironmentService environmentService, ConversionService conversionService, CredentialService credentialService, NetworkService networkService) {
+        this.environmentDtoConverter = environmentDtoConverter;
+        this.environmentRepository = environmentRepository;
+        this.environmentService = environmentService;
+        this.conversionService = conversionService;
+        this.credentialService = credentialService;
+        this.networkService = networkService;
+    }
+
+    public DetailedEnvironmentResponse edit(String environmentName, EnvironmentEditDto editDto) {
```

Comment:
This should return an EnvironmentDto and to be converted into a response in the Controller layer. Services msut be independent of the API layer!

Reply:
done. also changed the `changeCredential` method to return dto instead of response